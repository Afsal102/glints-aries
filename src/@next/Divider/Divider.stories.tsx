import { Meta, Story } from '@storybook/react';
import React from 'react';

import { BaseContainer } from '../../Layout/GlintsContainer/GlintsContainer';
import { Divider } from './Divider';

(Divider as React.FunctionComponent<void>).displayName = 'Divider';

export default {
  title: '@next/Divider',
  component: Divider,
  decorators: [Story => <BaseContainer>{Story()}</BaseContainer>],
} as Meta;

const Template: Story<void> = () => (
  <div>
    some text
    <Divider />
    another text after divider
  </div>
);

export const Interactive = Template.bind({});

Interactive.args = {};
