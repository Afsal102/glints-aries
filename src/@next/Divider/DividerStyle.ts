import styled from 'styled-components';
import { Neutral } from '../utilities/colors';
import { DividerProps } from './Divider';

export const DividerStyle = styled.div<DividerProps>`
  margin: 0;
  background-color: ${Neutral.B85};
  width: 100%;
  height: 1px;
`;
