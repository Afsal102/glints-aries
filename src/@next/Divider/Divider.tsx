import React from 'react';
import { DividerStyle } from './DividerStyle';

export type DividerProps = React.HTMLAttributes<HTMLDivElement>;

export const Divider = ({ ...props }: DividerProps) => (
  <DividerStyle {...props} />
);
