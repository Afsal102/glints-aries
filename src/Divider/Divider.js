/* @flow */
import React from 'react';
import { DividerWrapper } from '../Style/DividerStyle';

const Divider = (props: Props) => {
  const { theme, size, thickness } = props;

  return (
    <DividerWrapper theme={theme} size={size} thickness={thickness} />
  );
}

export default Divider;