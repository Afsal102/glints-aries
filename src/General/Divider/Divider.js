/* @flow */

import React from 'react';
import { DividerContainer } from '../../Style/DividerStyle';

const Divider = (props: Props) => {
  const {
    theme,
    className,
  } = props;

  return (
    <DividerContainer
      className={className}
      theme={theme}
    />
  );
};

type Props = {
  theme: string,
  className: string,
}

export default Divider;
