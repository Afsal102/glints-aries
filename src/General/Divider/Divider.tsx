import * as React from 'react';
import { DividerContainer } from '../../Style/General/DividerStyle';

const Divider: React.FunctionComponent<Props> = (props) => {
  const {
    theme,
    className,
    style,
  } = props;

  return (
    <DividerContainer
      id="aries-divider"
      className={className}
      style={style}
      theme={theme}
    />
  );
};

interface Props {
  theme?: string;
  className?: string;
  style?: React.CSSProperties;
}

export default Divider;
