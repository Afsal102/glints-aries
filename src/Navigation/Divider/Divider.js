/* @flow */

import React from 'react';
import { DividerWrapper } from '../../Style/DividerStyle';

const Divider = (props: Props) => {
  const {
    theme,
    className,
  } = props;

  return (
    <DividerWrapper
      className={className}
      theme={theme}
    />
  );
};

type Props = {
  theme: string,
  className: string,
}

export default Divider;
