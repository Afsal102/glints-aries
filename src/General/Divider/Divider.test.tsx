import * as React from 'react';
import 'jest-styled-components';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';

import Divider from './Divider';

describe('<Divider/> snapshots', () => {
  test('should match snapshot when no props is being passed', () => {
    const { asFragment } = render(<Divider />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('should match snapshot when all props is being passed', () => {
    const { asFragment } = render(
      <Divider theme="blue" className="name" style={{ top: '10px' }} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
