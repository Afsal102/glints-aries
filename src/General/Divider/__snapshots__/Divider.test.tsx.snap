// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Divider/> snapshots should match snapshot when no props is being passed 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 0.15em;
  background: #000000;
}

<div
    class="c0 aries-divider"
  />
</DocumentFragment>
`;

exports[`<Divider/> snapshots with theme prop theme blue 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 0.15em;
  background: #0BAEEC;
}

<div
    class="c0 aries-divider name"
    style="top: 10px;"
  />
</DocumentFragment>
`;

exports[`<Divider/> snapshots with theme prop theme default 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 0.15em;
  background: #000000;
}

<div
    class="c0 aries-divider name"
    style="top: 10px;"
  />
</DocumentFragment>
`;

exports[`<Divider/> snapshots with theme prop theme grey 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 0.15em;
  background: #C6C6C6;
}

<div
    class="c0 aries-divider name"
    style="top: 10px;"
  />
</DocumentFragment>
`;

exports[`<Divider/> snapshots with theme prop theme red 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 0.15em;
  background: #EC272B;
}

<div
    class="c0 aries-divider name"
    style="top: 10px;"
  />
</DocumentFragment>
`;

exports[`<Divider/> snapshots with theme prop theme white 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 0.15em;
  background: #FFFFFF;
}

<div
    class="c0 aries-divider name"
    style="top: 10px;"
  />
</DocumentFragment>
`;

exports[`<Divider/> snapshots with theme prop theme yellow 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  height: 0.15em;
  background: #FFF240;
}

<div
    class="c0 aries-divider name"
    style="top: 10px;"
  />
</DocumentFragment>
`;
